### ISSUE TITLE
[http] should we add @error for 4xx responses as error by default?

### ISSUE THREAD

**Issue by MaryGao** [microsoft/typespec#5315]
According to [the spec](https://www.w3.org/Protocols/HTTP/HTRESP.html), the 4xx codes are intended for cases in which the client seems to have erred so I think most cases the 4xx return should be treated as error responses. 

In our http spec the pre-defined responses like `NotFoundResponse` is NOT an error([definition here](https://github.com/microsoft/typespec/blob/main/packages/http/lib/http.tsp#L80-L96)). If we have usuage like below client side would take both 2xx + 404 as success codes, while 404 are actually errors and see [fix pr](https://github.com/allenjzhang/typespec-e2e-demo/pull/19).

[todo spec link](https://github.com/allenjzhang/typespec-e2e-demo/blob/b0e93c66e9902bf5f03e5e79b9afdcfb781ff2b9/todoApp/spec/main.tsp#L254)
```
 @get op get(@path id: TodoItem.id): TodoItem | NotFoundResponse;
  
 @delete op delete(
    @path id: TodoItem.id,
  ): WithStandardErrors<NoContentResponse | NotFoundResponse>;
```

This may not conform to general customers' cases and may introduce un-intentional behaviors. I was wondering in http spec side, if we could treate these 4xx as error by default? Or we could both provide NotFoundResponse and NotFoundErrorResponse for customers?

**Comment by lmazuel** [comment]
@MaryGao , errors needs to be tagged `@error`. If they're not, SDK client should consider this a success.

As we discussed in the example from todo you're using, we should update that TypeSpec to use `@error`.

My two cents would be to close that issue with "by design", but should be @bterlson call :)

**Comment by markcowl** [comment]
I think the intention is clear - if a response has `@error` it is an error, if it does not, it should be treated as a normal response.  In the case of this spec, I would agree that we should add `@error`.

**Comment by MaryGao** [comment]
Yeah, I understand errors needs to be tagged error decorator. What I am trying to express is since most cases 4XX would be an error, should we add @error decorator by default in http lib? 

```tsp
@error
model NotFoundResponse is Response<404>;
```

**Comment by markcowl** [comment]
won't fix
